src/control/server/mgmt_svc_test.go

```diff
@@ -139,6 +140,233 @@ func TestCheckMgmtSvcReplica(t *testing.T) {
 	}
 }
 
+func TestMgmtSvc_PoolCreate(t *testing.T) {
+	missingSB := newTestMgmtSvc(nil)
+	missingSB.harness.instances[0]._superblock = nil
+	notAP := newTestMgmtSvc(nil)
+	notAP.harness.instances[0]._superblock.MS = false
+
+	for name, tc := range map[string]struct {
+		mgmtSvc       *mgmtSvc
+		setupMockDrpc func(_ *mgmtSvc, _ error)
+		targetCount   int
+		req           *mgmtpb.PoolCreateReq
+		expResp       *mgmtpb.PoolCreateResp
+		expErr        error
+	}{
+		"nil request": {
+			expErr: errors.New("nil request"),
+		},
+		"missing superblock": {
+			mgmtSvc:     missingSB,
+			targetCount: 8,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  100 * humanize.GiByte,
+				Nvmebytes: 10 * humanize.TByte,
+			},
+			expErr: errors.New("not an access point"),
+		},
+		"not access point": {
+			mgmtSvc:     notAP,
+			targetCount: 8,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  100 * humanize.GiByte,
+				Nvmebytes: 10 * humanize.TByte,
+			},
+			expErr: errors.New("not an access point"),
+		},
+		"dRPC send fails": {
+			targetCount: 8,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  100 * humanize.GiByte,
+				Nvmebytes: 10 * humanize.TByte,
+			},
+			expErr: errors.New("send failure"),
+		},
+		"zero target count": {
+			targetCount: 0,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  100 * humanize.GiByte,
+				Nvmebytes: 10 * humanize.TByte,
+			},
+			expErr: errors.New("zero target count"),
+		},
+		"garbage resp": {
+			targetCount: 8,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  100 * humanize.GiByte,
+				Nvmebytes: 10 * humanize.TByte,
+			},
+			setupMockDrpc: func(svc *mgmtSvc, err error) {
+				// dRPC call returns junk in the message body
+				badBytes := makeBadBytes(42)
+
+				setupMockDrpcClientBytes(svc, badBytes, err)
+			},
+			expErr: errors.New("unmarshal"),
+		},
+		"successful creation": {
+			targetCount: 8,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  100 * humanize.GiByte,
+				Nvmebytes: 10 * humanize.TByte,
+			},
+			expResp: &mgmtpb.PoolCreateResp{},
+		},
+		"successful creation minimum size": {
+			targetCount: 8,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  128 * humanize.MiByte,
+				Nvmebytes: 8 * humanize.GByte,
+			},
+			expResp: &mgmtpb.PoolCreateResp{},
+		},
+		"failed creation scm too small": {
+			targetCount: 8,
+			req: &mgmtpb.PoolCreateReq{
+				Scmbytes:  127 * humanize.MiByte,
+				Nvmebytes: 8 * humanize.GByte,
+			},
+			expErr: FaultPoolScmTooSmall(127*(1<<20), 8),
+		},
```

Comment:
I think it would be clearer and more maintainable to express these as values relative to the consts, e.g.

```suggestion
		"failed creation scm too small": {
			targetCount: 8,
			req: &mgmtpb.PoolCreateReq{
				Scmbytes:  minScmSize - 1,
				Nvmebytes: minNvmeBytes,
			},
			expErr: FaultPoolScmTooSmall(minScmBytes-1, 8),
		},
```